source/heatfluxClass.py

```diff
@@ -554,6 +555,34 @@ def multiExp_profile_fluxspace(self, PFC, R, Bp, psiN, mode):
         q[nan_locations] = 0.0
         q[inf_locations] = 0.0
         return q
+    
+    def from_rzq_profile(self, rzq_data, PFC):
+        """"
+        q from any R(m), Z(m), q(W/m2) data
+        q is mapped from the R,Z grid to OMP
+        """
+
+        R = rzq_data['R(m)'].to_numpy()
+        Z = rzq_data['Z(m)'].to_numpy()
+        Q = rzq_data['q(W/m2)'].to_numpy()
+        # make Q positive. Sometimes, the q values from SOLPS is negative to indicate direction.
+        for i in range(0, len(Q)):
+            if Q[i] <0:
+                Q[i] *= -1
+        psi_rzq = PFC.ep.psiFunc.ev(R,Z) #convert (r,z) coordinate to psi
+        psi_rzq_omp = self.map_R_psi(psi_rzq, PFC) #map the psi_rzq to OMP
+
+        # print("psi_rzq is:", psi_rzq)
+        # print("psi_rzq_omp is", psi_rzq_omp)
+
+        q_interp = scinter.UnivariateSpline(psi_rzq_omp, Q, s = 0, ext = 'const') #interpolate the value of q at OMP
+        psi = self.map_R_psi(PFC.psimin, PFC) #map PFC centers to OMP
+        q1 = q_interp(psi) #calculate the value of q basesd on mapped to OMP PFC centers
+        q1 = q1/1e6
```

Comment:
this looks like you assume MW as input, which is not consistent with comment

Reply:
input is W/m2 but output is MW/m2 to be consistent with HEAT output